### PR TITLE
[MM-27967] api4/config_test: fix flaky test if siteurl is already set

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
If `MM_SERVICESETTINGS_SITEURL` is set in evironment variables, the test fails on updating to the default values since the condition in `api4/config` is not satisfied. (`*old.ServiceSettings.SiteURL == "" || *new.ServiceSettings.SiteURL != ""`) . We can pass an empty string to be able to set default values in the test.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-27967

### CODE REVIEW

**Review by mgdelacroix — APPROVED** [review]
LGTM :+1:

File: api4/config_test.go [hunk 1]
```diff
@@ -146,7 +145,7 @@ func TestUpdateConfig(t *testing.T) {
 		require.Equal(t, SiteName, cfg.TeamSettings.SiteName, "It should update the SiteName")
 
 		t.Run("Should set defaults for missing fields", func(t *testing.T) {
-			_, appErr := th.SystemAdminClient.DoApiPut(th.SystemAdminClient.GetConfigRoute(), fmt.Sprintf(`{"ServiceSettings":{"SiteURL":"%s"}}`, *cfg.ServiceSettings.SiteURL))
+			_, appErr := th.SystemAdminClient.DoApiPut(th.SystemAdminClient.GetConfigRoute(), `{"ServiceSettings":{"SiteURL":""}}`)
```
Comment (streamer45) [inline api4/config_test.go:148]: Wouldn't this cause a failure if the `SiteURL` is set? 
https://github.com/mattermost/mattermost-server/blob/f4ccc7061ca01ad41f3789d7d01443b2161f1c66/api4/config.go#L111-L114

Comment (isacikgoz) [reply api4/config_test.go:148]: Yeah you're right. This should be other-way round. It turned out that my env var was being set after the config initializes. Therefore it passed my test suite 😓

**Review by streamer45 — APPROVED** [review]
:+1:

**Pull request merged**